src/Reopt/Events.hs

```diff
@@ -348,15 +348,15 @@ printLogEvent ::
 printLogEvent event = do
   case event of
     ReoptGlobalStepStarted s ->
-      hPutStrLn stderr $ ppGlobalStep s
-    ReoptGlobalStepFinished _ _ ->
-      hPutStrLn stderr $ printf "  Complete."
+      hPutStrLn stderr $ printf "  [BEGIN] %s" (ppGlobalStep s)
+    ReoptGlobalStepFinished s _ ->
+      hPutStrLn stderr $ printf "  [ END ] %s." (ppGlobalStep s)
     ReoptGlobalStepWarning _st msg ->
-      hPutStrLn stderr $ printf "  %s" msg
+      hPutStrLn stderr $ printf "  [ WARN] %s" msg
```

Comment:
I wonder if it would be worth having top-level `warn`, `begin`, `end`, etc. functions to print a line of output with these heralds. Right now, this function would be the only call site, but I could see these being useful in other contexts later.

Reply:
Yeah. Ideally I'd also like to use an actual logging library (something like Kevin's lumberjack), might try to do the change on my own time.
But as a quick fix, I could at least extract these indeed.

Reply:
How does this look?
1e0741b2d344664fce94b10e95ed0cf1ad84c8d4

Reply:
Hmm I could have folded in the hPutStrLn call too.